### PR TITLE
fix(pom): don't remove excluded deps from upper pom's

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Fixed cases where we were removing excluded deps from upper pom's.
Example - https://github.com/aquasecurity/trivy/issues/5827
Test to better understand the fix - https://github.com/aquasecurity/go-dep-parser/blob/bf1b2d1cb5f5ad139c03d378674969ecd410d335/pkg/java/pom/parse_test.go#L689-L725

### CODE REVIEW

File: pkg/java/pom/pom.go [hunk 1]
```diff
@@ -4,6 +4,7 @@ import (
 	"encoding/xml"
 	"fmt"
 	"io"
+	"maps"
```
Comment (DmitriyLewen) [inline pkg/java/pom/pom.go:7]: Wait #283

Comment (DmitriyLewen) [reply pkg/java/pom/pom.go:7]: #283 has been merged.

File: pkg/java/pom/pom.go [hunk 2]
```diff
@@ -253,10 +254,12 @@ func (d pomDependency) Resolve(props map[string]string, depManagement, rootDepMa
...
-func (d pomDependency) ToArtifact(exclusions map[string]struct{}) artifact {
-	if exclusions == nil {
-		exclusions = map[string]struct{}{}
-	}
+func (d pomDependency) ToArtifact(ex map[string]struct{}) artifact {
+	// To avoid shadow adding exclusions to top pom's,
+	// we need to initialize a new map for each new artifact
+	// See `exclusions in child` test for more information
+	exclusions := map[string]struct{}{}
+	maps.Copy(exclusions, ex)
```
Comment (knqyf263) [inline pkg/java/pom/pom.go:262]: How about maps.Clone?

```suggestion
	exclusions := maps.Clone(ex)
```

Comment (DmitriyLewen) [reply pkg/java/pom/pom.go]: Looks like we will have enough of a hallow clone. Thank you!
Changed in https://github.com/aquasecurity/go-dep-parser/pull/282/commits/cdc2c7f5f7e7e092457c0c18762b35339495b7cf

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**